jupyterhub/templates/hub/configmap.yaml

```diff
@@ -5,312 +5,18 @@ metadata:
-  {{- end }}
-
-  {{- /* KubeSpawner */}}
-  kubespawner.common-labels: |
-    {{- $_ := merge (dict "heritageLabel" "jupyterhub") . }}
-    {{- include "jupyterhub.commonLabels" $_ | nindent 4 }}
-
-
-  {{- /* Hub */}}
-  hub.allow-named-servers: {{ .Values.hub.allowNamedServers | quote }}
-  hub.concurrent-spawn-limit: {{ .Values.hub.concurrentSpawnLimit | quote }}
-  hub.consecutive-failure-limit: {{ .Values.hub.consecutiveFailureLimit | quote }}
-  {{- if .Values.hub.activeServerLimit }}
-  hub.active-server-limit: {{ .Values.hub.activeServerLimit | quote }}
-  {{- end }}
-  {{- if .Values.hub.services }}
-  hub.services: |
-    {{- /* include hub.services *except* api_tokens */}}
-    {{- range $name, $service := .Values.hub.services }}
-    {{ $name }}:
-      {{- range $key, $value := $service }}
-      {{- if ne $key "apiToken" }}
-      {{- /* Use toJson since values can be more than just strings */}}
-      {{ $key }}: {{ toJson $value }}
-      {{- end }}
-      {{- end }}
-    {{- end }}
-  {{- end }}
-  {{- if .Values.hub.extraConfig }}
-  {{- $extraConfigType := typeOf .Values.hub.extraConfig }}
-  {{- if eq $extraConfigType "string" }}
-  hub.extra-config.default.py: |
-    {{- .Values.hub.extraConfig | nindent 4 }}
-  {{- else if eq $extraConfigType "map[string]interface {}" }}
-  {{- range $key, $value := .Values.hub.extraConfig }}
-  hub.extra-config.{{ $key }}.py: |
-    {{- $value | nindent 4 }}
-  {{- end }}
-  {{- end }}
-  {{- end }}
-  {{- if .Values.hub.extraConfigVars }}
-  hub.extra-vars: {{ toJson .Values.hub.extraConfigVars | quote }}
-  {{- end }}
-  hub.base_url: {{ .Values.hub.baseUrl | quote }}
-  {{- if eq .Values.hub.db.type "sqlite-pvc" }}
-  hub.db_url: "sqlite:///jupyterhub.sqlite"
-  {{- else if eq .Values.hub.db.type "sqlite-memory" }}
-  hub.db_url: "sqlite://"
-  {{- else }}
-  hub.db_url: {{ .Values.hub.db.url | quote }}
-  {{- end }}
-  {{- if .Values.hub.cookieSecret }}
-  token.cookie_secret: {{ .Values.hub.cookieSecret | quote }}
-  {{- end }}
-
-  {{- /* Other */}}
-  debug.enabled: {{ .Values.debug.enabled | quote }}
-  {{- range $key, $value := .Values.hub.extraConfigMap }}
-  custom.{{ $key }}: {{ toJson $value | quote }}
-  {{- end }}
+{{- $values := merge dict .Values -}}
+{{- /* trim secret values. Update here if new secrets are added! */ -}}
+{{- $_ := set $values "proxy" (omit $values.proxy "secretToken" "https.manual") -}}
+{{- $_ := set $values.proxy "https" (omit $values.proxy.https "manual") -}}
+{{- $_ := set $values "hub" (omit $values.hub "cookieSecret") -}}
+{{- /* make a copy of values.auth to avoid modifying the original */ -}}
+{{- $_ := set $values "auth" (merge dict .Values.auth) -}}
+{{- $_ := set $values.auth "state" (omit $values.auth.state "cryptoKey") -}}
+{{- range $key, $service := $values.hub.services -}}
+  {{- if $service.apiToken -}}
+    {{- $_ := set $values.hub.services $key (omit $service "apiToken") -}}
+  {{- end -}}
+{{- end }}
```

Comment:
Remove all right whitespace chomping here, if not you will have invalid output because everything will glue together on one line.

My practice throughout so far is to have right whitespace chomping on all statements until the first output in a .yaml file or helper, then only left whitespace chomping.

In this PR TODO:
- From line 8-19, delete the right whitespace chomping.

For me TODO:
- Make a templates styleguide

Reply:
> Remove all right whitespace chomping here, if not you will have invalid output because everything will glue together on one line.

This is not true. The whitespace chomping here does nothing other than ensure that all of these lines that produce no output at all actually produce no output.

> Line 18 cant have left whitespace chomping either, it is actual content.

This is also not true. None of these chomped lines produce output.

> Since you are not whitespace chomping left on line 18, this line will be indented now forcing similar indentation to happen on neighboring YAML content.

I think this is also inaccurate. These lines are pure logic (they produce no output). In general, indenting logic *should* be done, not avoided, to improve readability. Only where complex whitespace chomping causes a problem (i.e. not in this case) should it be avoided.

> Make a templates styleguide

That would be helpful, but more important, I think, is to be less strict about these things. Strict enforcement of style without an easy auto-formatter only creates an unnecessarily hostile contribution environment.

Reply:
Whitespace chomping left / right influences whitespace content left and right, nothing about the actual output of the template itself. Assignments `{{ $something := ... }} won't generate any output though, neither will the helm template comments `{{/* */}}`.

Ah I get it now, it was only assignments within the range. I got to thinking you were generating content in the `range` on line 17-19, but you were just iterating assignments. All generated content is formatted properly on line 22.

The enforcing to not generate output, works with only left whitespace chomping though, which is what I tend to do. Right whitespace chomping can mess things up, and is mainly done for linting pleasing. If you would had generated content in that range loop though, that content would have ended up right after `data:`.

We could go with "always left whitespace chomp, never right" as a rule. Together with usage of nindent I think that may be all we need to generate valid output

Reply:
Sorry for the misguided conclusions @minrk

Reply:
Right chomping removed, though it makes no difference in the output.

Reply:
I got a bad conscience about these comments, I have been too nitty, the linting of rendered output influenced me in a bad way. I'll fix it!

```
$ humctl patch human erik --patch '{"spec": {"nittyness": 0.1}}'

# Human          # Status
erik-61v8zlh     Terminating
erik-c6l1z9q     Pending
```

```
$ humctl get humans

# Human          # Status
erik-c6l1z9q     Running
```